internal/engine/dockerprune/docker_pruner.go

```diff
@@ -203,25 +202,85 @@ func (dp *DockerPruner) deleteOldImages(ctx context.Context, maxAge time.Duratio
 					inspect.ID, strings.Join(inspect.RepoTags, ", "))
 				continue
 			}
-			toDelete[inspect.ID] = uint64(inspect.Size)
+			result = append(result, inspect)
+		}
+	}
+	return result
+}
+
+// Return all image objects that aren't in the N
+// most recently used for each tag.
+func (dp *DockerPruner) keepMostRecentInspects(ctx context.Context, inspects []types.ImageInspect, keepRecent int, selectors []container.RefSelector) []types.ImageInspect {
+	// First, sort the images in order from most recent to least recent.
+	recentFirst := append([]types.ImageInspect{}, inspects...)
+	sort.SliceStable(recentFirst, func(i, j int) bool {
+		return recentFirst[i].Metadata.LastTagTime.After(recentFirst[j].Metadata.LastTagTime)
```

Comment:
not a big deal (I know it wasn't in the original) but should we add a note why we're using `LastTagTime` instead of `CreatedAt`? Makes sense once i thought of it but wasn't immediately obvious.

Reply:
done